source/include/FreeRTOSIPConfigDefaults.h

```diff
@@ -25,1139 +25,3368 @@
-#ifndef ipconfigSUPPORT_NETWORK_DOWN_EVENT
-    #define ipconfigSUPPORT_NETWORK_DOWN_EVENT    0
+/*---------------------------------------------------------------------------*/
+
+/*
+ * ipconfigWATCHDOG_TIMER
+ *
+ * https://www.freertos.org/FreeRTOS-Plus/FreeRTOS_Plus_TCP/TCP_IP_Configuration.html#ipconfigWATCHDOG_TIMER
+ *
+ * Type: Macro Function
+ *
+ * Macro that is called on each iteration of the IP task.
+ *
+ * May be useful if the application includes watchdog type functionality that
+ * needs to know that the IP task is still cycling (although the fact that the
+ * IP task is cycling does not necessarily indicate it is functioning
+ * correctly). The return value is ignored.
+ */
+
+#ifndef ipconfigWATCHDOG_TIMER
+    #define ipconfigWATCHDOG_TIMER()    do {} while( pdFALSE )
 #endif
 
-#ifndef ipconfigND_CACHE_ENTRIES
-    #define ipconfigND_CACHE_ENTRIES    24
+/*---------------------------------------------------------------------------*/
+
+/*
+ * ipconfigUSE_DUMP_PACKETS
+ *
+ * Type: BaseType_t ( ipconfigENABLE | ipconfigDISABLE )
+ *
+ * See this utility: tools/tcp_utilities/tcp_dump_packets.md
+ *
+ * Allow inclusion of a utility that writes of network packets to files.
+ *
+ * Useful for testing and development. Assumes the presence of full stdio
+ * disk access.
+ */
+
+#ifndef ipconfigUSE_DUMP_PACKETS
+    #define ipconfigUSE_DUMP_PACKETS    ipconfigDISABLE
 #endif
 
-#ifndef ipconfigHAS_ROUTING_STATISTICS
-    #define ipconfigHAS_ROUTING_STATISTICS    1
+#if ( ( ipconfigUSE_DUMP_PACKETS != ipconfigDISABLE ) && ( ipconfigUSE_DUMP_PACKETS != ipconfigENABLE ) )
+    #error Invalid ipconfigUSE_DUMP_PACKETS configuration
 #endif
 
-#ifndef ipconfigUSE_RA
-    #define ipconfigUSE_RA    1
+/*---------------------------------------------------------------------------*/
+
+/*
+ * ipconfigUSE_TCP_MEM_STATS
+ *
+ * Type: BaseType_t ( ipconfigENABLE | ipconfigDISABLE )
+ *
+ * See this utility: tools/tcp_utilities/tcp_mem_stats.md
+ *
+ * Allow inclusion of a utility that monitors all allocation and releases of
+ * network-related resources.
+ *
+ * After running for a while, it will print all data
+ * in a CSV format, which can be analysed in a spreadsheet program.
+ */
+
+#ifndef ipconfigUSE_TCP_MEM_STATS
+    #define ipconfigUSE_TCP_MEM_STATS    ipconfigDISABLE
+#endif
+
+#if ( ( ipconfigUSE_TCP_MEM_STATS != ipconfigDISABLE ) && ( ipconfigUSE_TCP_MEM_STATS != ipconfigENABLE ) )
+    #error Invalid ipconfigUSE_TCP_MEM_STATS configuration
 #endif
 
-/* RA or Router Advertisement/SLAAC: see end-point flag 'bWantRA'.
- * An Router Solicitation will be sent. It will wait for ipconfigRA_SEARCH_TIME_OUT_MSEC ms.
- * When there is no response, it will be repeated ipconfigRA_SEARCH_COUNT times.
- * Then it will be checked if the chosen IP-address already exists, repeating this
- * ipconfigRA_IP_TEST_COUNT times, each time with a timeout of ipconfigRA_IP_TEST_TIME_OUT_MSEC ms.
- * Finally the end-point will go in the UP state.
```

Comment:
I see a lot of such comments are being removed, can we add with the update as well, it will be useful for the end users to understand the macro better. Let me know what do you think about it.